source/common/router/BUILD

```diff
@@ -431,7 +431,14 @@ envoy_cc_library(
         "//source/common/protobuf:utility_lib",
         "@envoy_api//envoy/config/core/v3:pkg_cc_proto",
         "@envoy_api//envoy/config/route/v3:pkg_cc_proto",
-    ],
+    ] + select({
+        "//bazel:disable_static_extension_registration": [],
+        "//conditions:default": [
+            # See the comments in the header file.
+            "//source/common/formatter:formatter_extension_lib",
+            "//source/common/formatter:substitution_formatter_lib",
+        ],
+    }),
```

Comment:
This actually make thing too complex. basically, you only need to link the `substitution_formatter_lib` here which is exception free.

The main lib will link the `formatter_extension_lib` as extensions.